### ISSUE TITLE
Feature request: Integration with BGP advertisements

### ISSUE THREAD

**Issue by racompton** [NLnetLabs/krill#221]
Add in a BGP speaker into Krill or ingest BMP data to get the advertisements that an entity is announcing to the Internet and use this for ROA creation.
One option would be to automatically create ROAs based on what an ASN is currently advertising.
Another option would be to warn the user that is creating a ROA that the proposed ROA would contradict what the ASN is currently announcing (eg. prefix length doesn't match, ASN doesn't match, etc.).

**Comment by timbru** [comment]
Ah yes, this is something we have been discussing as well!

We are working on release 0.6 right now, but for 0.7 we were planning on better guidance to operators when creating ROAs.

Our first take on it would be to use RIPE RIS to present users with an overview of announcements seen for their address space. In theory the RIPE RIS *external* view can include unwanted announcements (hijacks, things forgotten, things unintended), so operators can then make an active decision to authorise the announcements. We would show the impact of the ROAs on validity of the announcements. Similar to interfaces currently provided by RIRs.

Once this is in place though we can go further and add other sources of BGP information. If the source is internal and trusted one could even go as far as doing just-in-time (or well just-too-late-but-probably-good-enough) authorisations. BMP is an option, but I need to look at it more closely.

Probably we will end up with making some plugins around the existing (and to be extended) API for ROA management. Then we could have a plugin that can take BMP input and adapt ROAs. Or maybe have a plugin that takes some other output format from a specific router vendor. Similarly the API could be used proactively if one is automating their networks: you could extend your system to not just tell a router to do an announcement, but also send a message to Krill to adapt ROAs.

In short: we want to start small, but there are many possible improvements along this path. Very happy to hear from you and others what you believe would be most useful in your operations.
